File: library-api/src/main/java/com/library/libraryapi/dto/search/SearchResponse.java
package com.library.libraryapi.dto.search;

import java.util.List;

public record SearchResponse(
        String query,
        int count,
        List<GoogleBookResult> results
) {
    public static SearchResponse of(String query, List<GoogleBookResult> results) {
        return new SearchResponse(query, results.size(), results);
    }
}

File: library-flutter/lib/features/books/data/models/search_response.dart
import 'google_book_result.dart';

class SearchResponse {
  final String query;
  final int count;
  final List<GoogleBookResult> results;

  SearchResponse({required this.query, required this.count, required this.results});

  factory SearchResponse.fromJson(Map<String, dynamic> json) => SearchResponse(
        query: json['query'] as String,
        count: json['count'] as int,
        results: (json['results'] as List<dynamic>)
            .map((e) => GoogleBookResult.fromJson(e as Map<String, dynamic>))
            .toList(),
      );
}
